src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpTraceExporterHelperExtensions.cs

```diff
@@ -61,28 +62,40 @@ public static TracerProviderBuilder AddOtlpExporter(this TracerProviderBuilder b
 
             name ??= Options.DefaultName;
 
-            if (configure != null)
+            builder.ConfigureServices(services =>
             {
-                builder.ConfigureServices(services => services.Configure(name, configure));
-            }
+                if (configure != null)
+                {
+                    services.Configure(name, configure);
+                }
+
+                services.RegisterOptionsFactory(configuration => new SdkLimitOptions(configuration));
+                services.RegisterOptionsFactory(configuration => new OtlpExporterOptions(configuration));
```

Comment:
Just to clarify my understanding of things...

Let's say I set `OTEL_EXPORTER_OTLP_ENDPOINT=http://my-collector.com` and I also have some json config like:

```json
{
  "MyOtlpExporterOptions: {
     "Headers": "api-key=abc123" 
  }
}
```

Then I do this somewhere.

```csharp
builder.Services.Configure<OtlpExporterOptions>(builder.Configuration.GetSection("MyOtlpExporterOptions"));
```

Will I get the headers setting from the json and the endpoint setting from the environment variable?

Reply:
> Will I get the headers setting from the json and the endpoint setting from the environment variable?

Yes!